contracts/core/FreeBets/FreeBetsHolder.sol

```diff
@@ -341,6 +469,60 @@ contract FreeBetsHolder is Initializable, ProxyOwned, ProxyPausable, ProxyReentr
         emit SetSportsAMM(_sportsAMM);
     }
 
+    /// @notice sets the free bet expiration period
+    /// @param _freeBetExpirationPeriod the new free bet expiration period
+    function setFreeBetExpirationPeriod(uint _freeBetExpirationPeriod, uint _freeBetExpirationUpgrade) external onlyOwner {
+        freeBetExpirationPeriod = _freeBetExpirationPeriod;
+        if (_freeBetExpirationUpgrade == 0) {
+            freeBetExpirationUpgrade = block.timestamp;
+        } else {
+            freeBetExpirationUpgrade = _freeBetExpirationUpgrade;
+        }
+        emit SetFreeBetExpirationPeriod(_freeBetExpirationPeriod, _freeBetExpirationUpgrade);
+    }
+
+    /// @notice sets the free bet expiration for a user
+    /// @param _user the address of the user
+    /// @param _collateral the address of the collateral
+    /// @param _freeBetExpiration the new free bet expiration
+    function setUserFreeBetExpiration(address _user, address _collateral, uint _freeBetExpiration) external onlyOwner {
+        freeBetExpiration[_user][_collateral] = _freeBetExpiration;
+    }
+
+    /// @notice sets the users with free bet per collateral
+    /// @param _users the addresses of the users
+    /// @param _collateral the address of the collateral
+    function setUsersWithAlreadyFundedFreeBetPerCollateral(
+        address[] calldata _users,
+        address _collateral
+    ) external onlyOwner {
+        for (uint i = 0; i < _users.length; i++) {
+            if (!usersWithFreeBetPerCollateral[_collateral].contains(_users[i])) {
+                usersWithFreeBetPerCollateral[_collateral].add(_users[i]);
+            }
+        }
+    }
+
+    /* ========== INTERNAL FUNCTIONS ========== */
+
+    function _isFreeBetValid(
+        address _user,
+        address _collateral
+    ) internal view returns (bool isValid, uint timeToExpiration) {
+        if (supportedCollateral[_collateral] && balancePerUserAndCollateral[_user][_collateral] > 0) {
+            if (freeBetExpiration[_user][_collateral] == 0) {
+                timeToExpiration = freeBetExpirationUpgrade + freeBetExpirationPeriod > block.timestamp
```

Comment:
the code can be nicer if you introduce a variable `expirationDateToUse` and set it depending whether `freeBetExpiration[_user][_collateral]` is 0 or not.